### PR TITLE
Add tests for the command line interface

### PR DESCRIPTION
**Pull request by kyouko-taiga**
The PR refactors the CLI so that it can be tested with XCTest.

### CODE REVIEW

**Comment by kyouko-taiga** [comment]
@lucteo Tests are failing on macOS and Ubuntu because the clang cannot compile the generated C++. Could you have a look?

**Comment by kyouko-taiga** [comment]
I moved the test for the `--emit binary` option on another PR (#260), as it is blocked by the current state of the transpiler.

File: Tests/ValCommandTests/ValCommandTests.swift [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  }
+
+  func testNoInput() throws {
+    XCTAssertThrowsError(try ValCommand.parse([]))
+  }
+
+  func testRawAST() throws {
+    let input = try XCTUnwrap(
+      Bundle.module.url(forResource: "Success", withExtension: ".val", subdirectory: "Inputs"),
```
Comment (dabrahams) [inline Tests/ValCommandTests/ValCommandTests.swift:27]: I don't understand this incantation.  You could solve that by factoring it out of all the places it's repeated and giving that component a doc comment!

**Review by dabrahams — APPROVED** [review]
Very nice work overall; having tests for the CLI will make it lots easier for me to finish the pipeline work with confidence.  Some more comments inline.

File: Sources/ValCommand/ValCommand.swift [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+      channel.write("note: ", in: [.bold, .cyan])
+    case .warning:
+      channel.write("warning: ", in: [.bold, .yellow])
+    case .error:
+      channel.write("error: ", in: [.bold, .red])
+    }
+  }
+
+  /// Returns the path of the specified executable.
+  private func find(_ executable: String) throws -> String {
```
[the remark this answers is not in the document]
Comment (kyouko-taiga) [reply Sources/ValCommand/ValCommand.swift:372]: Good point. I'll make a separate PR.

File: Sources/ValCommand/ValCommand.swift [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+  }
+
+  /// Creates a module from the contents at `url` and adds it to the AST.
+  ///
+  /// - Requires: `url` must denote a directly.
+  private func addModule(url: URL) {
+    fatalError("not implemented")
+  }
+
+  /// Logs the contents of `diagnostics` tot he standard error.
```
[the remark this answers is not in the document]
Comment (kyouko-taiga) [reply Sources/ValCommand/ValCommand.swift:283]: Good point!

**Pull request merged**